tests/integration/test_binary_op_typing.py

```diff
@@ -104,31 +89,45 @@ def run_all_tests():
     # If anyone can do a better job than me and finally make cuNumeric
     # implement the same typing rules, please put these tests back.
     #
-    # for idx, lhs_type in enumerate(types):
-    #    for rhs_type in types[idx:]:
-    #        for array, scalar in product(array_values, scalar_values):
+    # for idx, lhs_type in enumerate(TYPES):
+    #    for rhs_type in TYPES[idx:]:
+    #        for array, scalar in product(ARRAY_VALUES, SCALAR_VALUES):
     #            try:
     #                lhs_np = np.array(array, dtype=lhs_type)
     #                rhs_np = np.array(scalar, dtype=rhs_type)
-
     #                lhs_num = num.array(lhs_np)
     #                rhs_num = num.array(rhs_np)
-
-    #                test(lhs_np, rhs_np, lhs_num, rhs_num)
+    #                yield (lhs_np, rhs_np, lhs_num, rhs_num)
     #            except TypeError:
     #                pass
 
     #            try:
     #                lhs_np = np.array(scalar, dtype=lhs_type)
     #                rhs_np = np.array(array, dtype=rhs_type)
-
     #                lhs_num = num.array(lhs_np)
     #                rhs_num = num.array(rhs_np)
-
-    #                test(lhs_np, rhs_np, lhs_num, rhs_num)
+    #                yield (lhs_np, rhs_np, lhs_num, rhs_num)
     #            except TypeError:
     #                pass
```

Comment:
I would suggest putting any remaining commented test code under `pytest.mark.skip` (I have already done so in some simple cases)

Reply:
Just FYI most of the array-scalar cases do pass:
```
2246 passed, 35 xfailed, 957 xpassed in 4.87s 
```
Do you want me to use `skip` or `xfail` here? It won't affect overall test passing either way, but with `xfail` the tests will run and report as above.